### PR TITLE
SPEC-818 auth spec tests

### PR DESCRIPTION
**Pull request by bjori**

### CODE REVIEW

**Comment by ajdavis** [comment]
LGTM.

**Comment by mbroadst** [comment]
LGTM, how'd that mocha get in there? :smile:

File: source/auth/tests/connection-string.json [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+			"valid": true,
+			"warning": false,
+			"auth": {
+				"username": "user",
+				"password": null
+			},
+			"options": null
+		},
+		{
+			"description": "should recognize no password",
```
Comment (behackett) [inline source/auth/tests/connection-string.json]: This is a duplicate of the test above it.

Comment (bjori) [reply source/auth/tests/connection-string.json]: Thank. I noticed that in the .js file too, but forgot to remove it! Fixed now

**Comment by behackett** [comment]
LGTM

**Pull request merged**